ee/hogai/llm.py

```diff
@@ -94,12 +109,32 @@ def _enrich_messages(self, messages: list[list[BaseMessage]], project_org_user_v
                     break
         return messages
 
+    def _with_posthog_properties(
+        self,
+        kwargs: Mapping[str, Any] | None = None,
+    ) -> dict[str, Any]:
+        """Return a shallow copy of kwargs with PostHog properties, billable flag, and team_id injected into metadata."""
+        new_kwargs = dict(kwargs or {})
+        metadata = dict(new_kwargs.get("metadata") or {})
+
+        # Build posthog_properties with billable flag and team_id
+        posthog_props = dict(self.posthog_properties or {})
+        posthog_props["$ai_billable"] = self.billable
+        posthog_props["team_id"] = self.team.id
```

Comment:
A couple of enhancements here (also not blocking, although once we settle on a convention, that'll be permanent):
1. Piggyback off of group analytics, i.e.`$groups` property
2. Include org ID, since it's free
```suggestion
        posthog_props["$groups"] = {"project": str(self.team.id), "organization": str(self.team.organization_id)}
```
([example of groups elsewhere](https://github.com/PostHog/posthog/blob/3224e517b1cca8d5c6a31dab622e92da31011ff2/posthog/hogql_queries/legacy_compatibility/feature_flag.py#L55-L58))

Reply:
Here it is not really needed at the trace level since we have that `key` already and we also have the `organization` val as well.

<img width="517" height="138" alt="image" src="https://github.com/user-attachments/assets/e30259a7-e127-4b12-b62b-8fa720c38f95" />